src/mem_table.rs

```diff
@@ -160,8 +170,15 @@ impl KVTable {
     /// Returns None if the key is not in the memtable at all,
     /// Some(None) if the key is in the memtable but has a tombstone value,
     /// Some(Some(value)) if the key is in the memtable with a non-tombstone value.
-    pub(crate) fn get(&self, key: &[u8]) -> Option<ValueWithAttributes> {
-        self.map.get(key).map(|entry| entry.value().clone())
+    pub(crate) fn get(&self, key: &[u8]) -> Option<RowEntry> {
+        // TODO: get the last element is not considered as efficient in the current SkipMap's code.
```

Comment:
the secondary sort should simply have reverse order, meaning the latest entry (= highest sequence number) for the key should come first; that's also the read order you'd expect during iteration - first you want to see latest updates for the key

Reply:
Also that's the order how entries should end up in SST files. An analogy to this is that you want latest data to end up in levels that you scan first (`WAL -> MEM -> L0 -> SR, ...`)

Reply:
thank you for the infomation, i'll reverse the sort order of the SkipMap in the later commit :)

Reply:
found SkipMap has a `next_back()` method which allows iterate backwards and it'll be more efficient than using `last()`, fixed it in 302d4a9

Reply:
Still, if you're iterating the memtable (range scan for example), shouldn't you always see the latest data first?

Reply:
Related discussion: https://github.com/slatedb/slatedb/pull/391#discussion_r1893239050

Reply:
yes, we should always take the latest data. 

it seems that the MergeIterator needs to be adjusted as well. It contains a heap inside it, and I guess we need to let the seqnum determine the ordering in this heap to deduplicate the data 😲

Reply:
i've added the deduplicated logic in this pr, still need append some tests for it!